src/stores/appcatalog/actions.ts

```diff
@@ -0,0 +1,33 @@
+import GiantSwarm from 'giantswarm';
+
+import { createAsynchronousAction } from '../asynchronousAction';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const listCatalogs = createAsynchronousAction<any, any, any>({
```

Comment:
These are PayloadType, StateType, and ResponseType.

I haven't made interfaces for that yet, mostly because StateType is the whole redux state, and I'm not ready to tackle that yet, and it would be out of scope of this PR.